airflow/providers/amazon/aws/example_dags/example_eks_with_fargate_profile.py

```diff
@@ -0,0 +1,124 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from datetime import datetime
+from os import environ
+
+from airflow.models.dag import DAG
+from airflow.providers.amazon.aws.hooks.eks import ClusterStates, FargateProfileStates
+from airflow.providers.amazon.aws.operators.eks import (
+    EKSCreateClusterOperator,
+    EKSCreateFargateProfileOperator,
+    EKSDeleteClusterOperator,
+    EKSDeleteFargateProfileOperator,
+    EKSPodOperator,
+)
+from airflow.providers.amazon.aws.sensors.eks import EKSClusterStateSensor, EKSFargateProfileStateSensor
+
+CLUSTER_NAME = 'fargate-demo'
+FARGATE_PROFILE_NAME = f'{CLUSTER_NAME}-profile'
+SELECTORS = environ.get('FARGATE_SELECTORS', [{'namespace': 'default'}])
+
+ROLE_ARN = environ.get('EKS_DEMO_ROLE_ARN', 'arn:aws:iam::123456789012:role/role_name')
+SUBNETS = environ.get('EKS_DEMO_SUBNETS', 'subnet-12345ab subnet-67890cd').split(' ')
+VPC_CONFIG = {
+    'subnetIds': SUBNETS,
+    'endpointPublicAccess': True,
+    'endpointPrivateAccess': False,
+}
+
+default_args = {
+    'cluster_name': CLUSTER_NAME,
+}
+
+with DAG(
+    dag_id='example_eks_with_fargate_profile_dag',
+    default_args=default_args,
```

Comment:
It's more readable to have the `cluster_name` arg directly set in the `DAG()` instantiation.  Another instance of something else being cleaned up from old DAG examples.  Sorry for the nit here.

```
with DAG(
    dag_id='example_eks_with_fargate_profile_dag',
    default_args={'cluster_name': CLUSTER_NAME},
  ....
)
```

Reply:
Does this work with templated values as well?  ie:  

```
with DAG(
    dag_id='example_eks_with_fargate_profile_dag',
    default_args={'cluster_name': "{{ dag_run.conf['cluster_role_arn'] }}"},
  ....
)
```

Reply:
Sure does!

Reply:
Should be addressed in https://github.com/apache/airflow/pull/18645/commits/b577f159075d1c6b75d90c54303041b78025f122